include/openenclave/bits/seal.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file seal.h
+ *
+ * This file defines constants and structures for sealing APIs.
+ *
+ * Only TEE agnostic definitions should go in this file. SGX specific
+ * definitions should go in sgx/seal.h
+ */
+
+#ifndef _OE_BITS_SEAL_H
+#define _OE_BITS_SEAL_H
+
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+typedef struct _oe_seal_setting
+{
+    int type;
+    uint32_t size;
+    union {
+        uint64_t q;
+        uint32_t d;
+        uint16_t w;
+        uint8_t b;
+        void* p;
+    } value;
+} oe_seal_setting_t;
+
+/**
+ * TEE agnostic seal settings supported by OE SDK
+ */
+typedef enum _oe_seal_setting_name
+{
+    OE_SEAL_SETTING_POLICY,
+    OE_SEAL_SETTING_ADDITIONAL_CONTEXT,
+    OE_SEAL_SETTING_IV,
+    OE_SEAL_SETTING_TEE_AGNOSTIC_MAX
+} oe_seal_setting_name_t;
+
+#define __OE_INIT_SEAL_SETTING_PTR(t, p, s) \
+    {                                       \
+        (t), s,                             \
+        {                                   \
+            (uint64_t)(p)                   \
+        }                                   \
+    }
+#define __OE_INIT_SEAL_SETTING(t, v) __OE_INIT_SEAL_SETTING_PTR(t, v, 0)
+#define OE_INIT_SEAL_POLICY(w) __OE_INIT_SEAL_SETTING(OE_SEAL_SETTING_POLICY, w)
+#define OE_INIT_SEAL_CONTEXT(p, s) \
+    __OE_INIT_SEAL_SETTING_PTR(OE_SEAL_SETTING_ADDITIONAL_CONTEXT, p, s)
+#define OE_INIT_SEAL_IV(p, s) \
+    __OE_INIT_SEAL_SETTING_PTR(OE_SEAL_SETTING_IV, p, s)
+
+/**
+ * Seal data to an enclave using AEAD (Authenticated Encryption with
+ * Additioonal Data).
+ *
+ * @param[in] plugin_id Optional UUID of the plugin to use. If \c NULL, the
+ * default plugin will be used.
+ * @param[in] settings Optional array of seal settings to be used.
+ * @param[in] settings_count The number of settings specified by \p settings.
+ * Must be \c 0 if \p settings is \c NULL.
+ * @param[in] plaintext Optional buffer to be encrypted under the seal key.
+ * @param[in] plaintext_size Size of \p plaintext, must be \c 0 if \p plaintext
+ * is \c NULL.
+ * @param[in] additional_data Optional additional data to be MAC'd under the
+ * @param[in] additional_data_size Size of \p additional_data, must be \c 0 if
+ * \p additional_data is \c NULL.
+ * seal key.
+ * @param[out] blob On success, receive the pointer to a buffer containing both
+ * \p additional_data and encrypted \p plaintext, along with necessary
```

Comment:
Is this comment accurate? I thought the new design is that the additional_data is not incorporated into the output blob.